cmd/api-linter/cli.go

```diff
@@ -181,6 +187,13 @@ func (c *cli) lint(rules lint.RuleRegistry, configs lint.Configs) error {
 	if _, err = w.Write(b); err != nil {
 		return err
 	}
+
+	// Return error on lint failure which subsequently
+	// exits with a non-zero status code
+	if c.ExitStatusOnLintFailure && len(results) > 0 {
```

Comment:
It should be checked if there are any problems in `results`:

```
results[*].Problems
```

Currently the linter returns a non-zero status code (false-positive).

Reply:
Thanks, my test cases weren't doing what I thought they were. I'll submit a fix for this